File: day06/part1.ts
import * as path from "path";
import { readFileSync } from "fs";

function compute(groups: String[]): number {
  return groups
    .map((group) => new Set(group.replace(/\n/g, "")).size)
    .reduce((a, b) => a + b);
}

function main() {
  const inputPath = path.resolve(__dirname, "input.txt");
  const input = readFileSync(inputPath, "utf-8");
  console.log(compute(input.split("\n\n")));
}

main();

File: day06/part1.py
_TESTCASE = """\
abc

a
b
c

ab
ac

a
a
a
a

b
""".strip().split(
    "\n\n"
)


def compute(data):
    """
    >>> compute(_TESTCASE)
    11
    """
    return sum(len(set(group) - {"\n"}) for group in data)


def main():
    import pathlib

    input_path = pathlib.Path(__file__).with_name("input.txt")

    with input_path.open() as f:
        print(compute(f.read().strip().split("\n\n")))


if __name__ == "__main__":
    main()
